cognite/client/data_classes/simulators/routine_revisions.py

```diff
@@ -666,6 +684,119 @@ def as_write(self) -> SimulatorRoutineRevisionWriteList:
         )
 
 
+class SimulatorRoutineStageList(UserList[SimulatorRoutineStage]):
+    """List of simulator routine stages with pandas conversion capabilities."""
+
+    def __init__(self, initlist: list[SimulatorRoutineStage] | None = None) -> None:
+        super().__init__(initlist)
+
+    def to_pandas(self) -> pandas.DataFrame:
+        """Convert the list of stages to a pandas DataFrame.
+
+        Returns:
+            pandas.DataFrame: DataFrame with stage and step information.
+        """
+        pd = local_import("pandas")
+        if not self.data:
+            return pd.DataFrame()
+
+        rows = []
+        for stage in self.data:
+            for step in stage.steps:
+                row = {
+                    "stage_order": stage.order,
+                    "stage_description": stage.description,
+                    "step_order": step.order,
+                    "step_type": step.step_type,
+                    "step_description": step.description,
+                }
+                for key, value in step.arguments.items():
+                    row[f"arg_{to_snake_case(key)}"] = value
+                rows.append(row)
+
+        return pd.DataFrame(rows)
+
+    def _repr_html_(self) -> str:
+        try:
+            return self.to_pandas()._repr_html_()
+        except Exception:
+            return str(self)
+
+
+class SimulatorRoutineInputList(UserList[SimulatorRoutineInput]):
+    """List of simulator routine inputs with pandas conversion capabilities."""
+
+    def __init__(self, initlist: list[SimulatorRoutineInput] | None = None) -> None:
+        super().__init__(initlist)
+
+    def to_pandas(self) -> pandas.DataFrame:
+        """Convert the list of inputs to a pandas DataFrame.
+
+        Returns:
+            pandas.DataFrame: DataFrame with input information.
+        """
+        pd = local_import("pandas")
+        if not self.data:
+            return pd.DataFrame()
+
+        rows = []
+        for item in self.data:
+            row = item.dump(camel_case=False)
+            if "unit" in row and row["unit"] is not None:
+                unit_data = row.pop("unit")
+                row["unit_name"] = unit_data.get("name")
+                row["unit_quantity"] = unit_data.get("quantity")
+            else:
+                row.pop("unit", None)
+                row["unit_name"] = None
+                row["unit_quantity"] = None
```

Comment:
```suggestion
            if (unit_data := row.pop("unit", None)) is not None:
                row["unit_name"] = unit_data.get("name")
                row["unit_quantity"] = unit_data.get("quantity")
            else:
                row["unit_name"] = None
                row["unit_quantity"] = None
```

Reply:
Fancy! Never uses `:=` before. I remember reading when this was included in a previous Python release, but didn't adopted it yet.